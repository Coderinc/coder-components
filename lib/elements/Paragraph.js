import PropTypes from 'prop-types';
import styled from 'styled-components';

const Paragraph = styled.p`
  color: ${({ theme }) => theme.primaryFont};
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: ${({ bold }) => (bold ? 'bold' : 'lighter')};
`;

Paragraph.propTypes = {
  bold: PropTypes.bool,
};

Paragraph.defaultProps = {
  bold: false,
};

export default Paragraph;
