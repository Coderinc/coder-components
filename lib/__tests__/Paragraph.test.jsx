import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Paragraph from '../elements/Paragraph';

describe('Paragraph', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Paragraph>Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('bold font', () => {
    const component = renderer.create(<Paragraph bold>Lorem Ipsum</Paragraph>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});