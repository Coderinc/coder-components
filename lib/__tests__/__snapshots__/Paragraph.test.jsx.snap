// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph bold font 1`] = `
.c0 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: bold;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;

exports[`Paragraph renders correctly 1`] = `
.c0 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: lighter;
}

<p
  className="c0"
>
  Lorem Ipsum
</p>
`;
